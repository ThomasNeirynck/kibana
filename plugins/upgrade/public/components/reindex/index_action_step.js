import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import {
  isStepCompleted,
  isStepFailed,
  isStepRunning,
  isStepCanceled,
  getStepMessage,
} from '../../lib';


export function IndexActionStep({ step, action, indexName }) {
  const iconClasses = classNames('kuiStatusText__icon', 'kuiIcon', {
    'kuiIcon--success fa-check': isStepCompleted(step) || isStepCanceled(step),
    'kuiIcon--error fa-warning': isStepFailed(step),
    'kuiIcon--basic fa-spinner fa-spin': isStepRunning(step),
  });

  return (
    <div className="kuiText">
      <div className="kuiStatusText">
        <span className={ iconClasses } />
        { getMessage(step, action, indexName) }
      </div>
    </div>
  );
}

IndexActionStep.propTypes = {
  step: PropTypes.object,
  action: PropTypes.string,
  indexName: PropTypes.string,
};

IndexActionStep.defaultProps = {
  steps: {},
  action: '',
  indexName: '',
};

function getMessage(step, action, indexName) {
  return getStepMessage(action, step.result, step.name, { ...step, indexName });
}
