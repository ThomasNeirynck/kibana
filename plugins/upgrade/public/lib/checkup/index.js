import {
  getFromApi,
} from '../request';


export async function getDeprecations() {
  return await getFromApi(`/api/migration/deprecations`);
}
