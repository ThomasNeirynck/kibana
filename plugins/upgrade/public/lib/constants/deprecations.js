export const DEPRECATION_ISSUE_LEVELS = [
  'critical',
  'warning',
  'info',
  'none',
];
