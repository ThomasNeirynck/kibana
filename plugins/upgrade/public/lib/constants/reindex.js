export const REINDEX_STEP_SEQUENCE = [
  'CREATE_INDEX',
  'SET_READONLY',
  'REINDEX',
  'REFRESH_INDEX',
  'VERIFY_DOCS',
  'REPLACE_INDEX',
];

export const REINDEX_STEPS = REINDEX_STEP_SEQUENCE
  .reduce((steps, step) => ({
    ...steps,
    [step]: step,
  }), {});

export const UPGRADE_STEP_SEQUENCE = [
  'UPGRADE',
];

export const UPGRADE_STEPS = UPGRADE_STEP_SEQUENCE
  .reduce((steps, step) => ({
    ...steps,
    [step]: step,
  }), {});

export const NUMBER_OF_STEPS = {
  REINDEX: REINDEX_STEP_SEQUENCE.length,
  UPGRADE: UPGRADE_STEP_SEQUENCE.length,
};

export const STEP_RESULTS = {
  COMPLETED:   'COMPLETED',
  FAILED:      'FAILED',
  RUNNING:     'RUNNING',
  NOT_STARTED: 'NOT_STARTED',
  CANCELED:    'CANCELED',
};

export const INDEX_ACTION = {
  TYPE: {
    REINDEX: 'REINDEX',
    UPGRADE: 'UPGRADE',
  },
  LABEL: {
    REINDEX: 'Reindex',
    UPGRADE: 'Upgrade',
  },
};
