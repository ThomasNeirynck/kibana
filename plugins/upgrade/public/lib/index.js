export { timeout } from './util';
export {
  getAssistance,
  getMappingsAndSettings,
  createIndex,
  setReadOnly,
  runReindex,
  getTaskDetails,
  refreshIndex,
  verifyDocs,
  getSettingsAndAliases,
  updateRefreshInterval,
  replaceIndex,
  deleteTask,
  resetIndex,
  cancelTask,
  runUpgrade,
  isCompleted,
  isFailed,
  isNotStarted,
  isRunning,
  isStepCompleted,
  isStepFailed,
  isStepNotStarted,
  isStepRunning,
  isResettable,
  getStepMessage,
  wrapErrorMessage,
} from './reindex';
