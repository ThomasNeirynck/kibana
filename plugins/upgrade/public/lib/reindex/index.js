export {
  isCompleted,
  isFailed,
  isNotStarted,
  isRunning,
  isCanceled,
  isResettable,
  isCancelable,
  isStepCompleted,
  isStepFailed,
  isStepNotStarted,
  isStepRunning,
  isStepCanceled,
} from './state';

export {
  getStepMessage,
  wrapErrorMessage,
} from './messages';

export {
  getAssistance,
  getMappingsAndSettings,
  getSettingsAndAliases,
  updateRefreshInterval,
  createIndex,
  setReadOnly,
  runReindex,
  getTaskDetails,
  cancelTask,
  refreshIndex,
  verifyDocs,
  replaceIndex,
  deleteTask,
  resetIndex,
  runUpgrade,
} from './step_methods';

