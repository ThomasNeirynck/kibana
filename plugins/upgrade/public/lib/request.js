import chrome from 'ui/chrome';

function get(url) {
  return new Promise((resolve, reject) => {
    const req = new window.XMLHttpRequest();

    req.open('GET', url);

    req.onload = () => {
      if (req.status === 200) {
        resolve(JSON.parse(req.response));
      } else {
        reject(JSON.parse(req.response));
      }
    };

    req.onerror = () => {
      reject(new Error('Network error'));
    };

    req.send();
  });
}

export function getFromApi(url) {
  return get(chrome.addBasePath(url));
}

function post(url, data) {
  return new Promise((resolve, reject) => {
    const xhr = new window.XMLHttpRequest();
    xhr.open('POST', url);
    xhr.setRequestHeader('Content-type', 'application/json');
    xhr.setRequestHeader('kbn-version', chrome.getXsrfToken());
    xhr.onload = () => {
      if (xhr.status === 200) {
        resolve(JSON.parse(xhr.response));
      } else {
        reject(JSON.parse(xhr.response));
      }
    };
    xhr.send(JSON.stringify(data));
  });
}

export function postToApi(url, data) {
  return post(chrome.addBasePath(url), data);
}

function put(url, data) {
  return new Promise((resolve, reject) => {
    const xhr = new window.XMLHttpRequest();
    xhr.open('PUT', url);
    xhr.setRequestHeader('Content-type', 'application/json');
    xhr.setRequestHeader('kbn-version', chrome.getXsrfToken());
    xhr.onload = () => {
      if (xhr.status === 200) {
        resolve(JSON.parse(xhr.response));
      } else {
        reject(JSON.parse(xhr.response));
      }
    };
    xhr.send(JSON.stringify(data));
  });
}

export function putToApi(url, data) {
  return put(chrome.addBasePath(url), data);
}

function del(url) {
  return new Promise((resolve, reject) => {
    const xhr = new window.XMLHttpRequest();
    xhr.open('DELETE', url);
    xhr.setRequestHeader('Content-type', 'application/json');
    xhr.setRequestHeader('kbn-version', chrome.getXsrfToken());
    xhr.onload = () => {
      if (xhr.status === 200) {
        resolve(JSON.parse(xhr.response));
      } else {
        reject(JSON.parse(xhr.response));
      }
    };
    xhr.send();
  });
}

export function deleteFromApi(url) {
  return del(chrome.addBasePath(url));
}
