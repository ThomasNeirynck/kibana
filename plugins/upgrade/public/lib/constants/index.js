import { HomeView } from '../../components/home';
import { CheckupView } from '../../components/checkup';
import { ReindexView } from '../../components/reindex';
import { LoggingView } from '../../components/logging';


export {
  NUMBER_OF_STEPS,
  REINDEX_STEPS,
  UPGRADE_STEPS,
  STEP_RESULTS,
  INDEX_ACTION,
} from './reindex';

export {
  DEPRECATION_ISSUE_LEVELS,
} from './deprecations';

export const VIEWS = {
  HOME: {
    component: HomeView,
    label: 'Overview',
    location: 'home',
  },
  CHECKUP: {
    component: CheckupView,
    label: 'Cluster Checkup',
    location: 'checkup',
  },
  REINDEX: {
    component: ReindexView,
    label: 'Reindex Helper',
    location: 'reindex',
  },
  LOGGING: {
    component: LoggingView,
    label: 'Toggle Deprecation Logging',
    location: 'logging',
  }
};
export const DEFAULT_VIEW_ID = 'HOME';

export const HUMAN_READABLE_DELAY = 750;

export const LOADING_STATUS = {
  FORBIDDEN: 'FORBIDDEN',
  FAILURE: 'FAILURE',
  LOADING: 'LOADING',
  SUCCESS: 'SUCCESS',
  UNINITIALIZED: 'UNINITIALIZED',
};
