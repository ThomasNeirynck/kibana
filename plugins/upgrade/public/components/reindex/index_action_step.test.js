import { shallow } from 'enzyme';
import React from 'react';

import { IndexActionStep } from './index_action_step';
import { getStepMessage } from '../../lib';
import { STEP_RESULTS } from '../../lib/constants';

jest.mock('ui/chrome', () => {}, { virtual: true });
jest.mock('ui/notify/notifier', () => ({
  Notifier: function Notifier() {},
}), { virtual: true });
jest.mock('../../lib/reindex/messages');


describe('IndexActionStep', () => {
  beforeEach(() => {
    getStepMessage.mockImplementation((action, result, name, { indexName }) => (
      `Message for ${action} step ${name} for index ${indexName} with result ${result}`
    ));
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  test('renders a success message for a completed step', () => {
    const component = (
      <IndexActionStep
        step={ {
          name: "COMPLETED STEP",
          result: STEP_RESULTS.COMPLETED,
        } }
        action="COMPLETED ACTION"
        indexName="COMPLETED INDEX_NAME"
      />
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders an error message for a failed step', () => {
    const component = (
      <IndexActionStep
        step={ {
          name: "FAILED STEP",
          result: {
            message: 'FAILED',
          },
        } }
        action="FAILED ACTION"
        indexName="FAILED INDEX_NAME"
      />
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders a spinner message for a running step', () => {
    const component = (
      <IndexActionStep
        step={ {
          name: "RUNNING STEP",
          result: STEP_RESULTS.RUNNING,
        } }
        action="RUNNING ACTION"
        indexName="RUNNING INDEX_NAME"
      />
    );

    expect(shallow(component)).toMatchSnapshot();
  });
});
