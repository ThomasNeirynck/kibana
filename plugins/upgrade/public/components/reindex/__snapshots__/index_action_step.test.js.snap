// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexActionStep renders a spinner message for a running step 1`] = `
<div
  className="kuiText"
>
  <div
    className="kuiStatusText"
  >
    <span
      className="kuiStatusText__icon kuiIcon kuiIcon--basic fa-spinner fa-spin"
    />
    Message for RUNNING ACTION step RUNNING STEP for index RUNNING INDEX_NAME with result RUNNING
  </div>
</div>
`;

exports[`IndexActionStep renders a success message for a completed step 1`] = `
<div
  className="kuiText"
>
  <div
    className="kuiStatusText"
  >
    <span
      className="kuiStatusText__icon kuiIcon kuiIcon--success fa-check"
    />
    Message for COMPLETED ACTION step COMPLETED STEP for index COMPLETED INDEX_NAME with result COMPLETED
  </div>
</div>
`;

exports[`IndexActionStep renders an error message for a failed step 1`] = `
<div
  className="kuiText"
>
  <div
    className="kuiStatusText"
  >
    <span
      className="kuiStatusText__icon kuiIcon kuiIcon--error fa-warning"
    />
    Message for FAILED ACTION step FAILED STEP for index FAILED INDEX_NAME with result [object Object]
  </div>
</div>
`;
